source/broadband/webpa_notification.c

```diff
@@ -282,6 +291,98 @@ void FactoryResetCloudSyncTask()
 		WalInfo("FactoryResetCloudSync Thread created Successfully\n");
 	}
 }
+void *SyncNotifyRetry()
+{
+	pthread_detach(pthread_self());
+	char *dbCMC = NULL;
+	int retryCount = 0;
+	int backoffRetryTime = 0;
+	int backoff_max_time = 9;
+	struct timespec ts;
+	int c = 3;
+	int  rv;    
+	int max_retry_sleep = (int) pow(2, backoff_max_time) -1;
+	WalInfo("max_retry_sleep is %d\n", max_retry_sleep );
+        
+	while(FOREVER())
+	{
+		if(backoffRetryTime < max_retry_sleep)
+       	{
+			backoffRetryTime = (int) pow(2, c) -1;
+		}
+		clock_gettime(CLOCK_REALTIME, &ts);
+		ts.tv_sec += backoffRetryTime;
+		//wait for backoff delay for retransmission
+		WalInfo("Wait for backoffRetryTime %d sec to receive cloud_conn_online event\n", backoffRetryTime);
+		rv = pthread_cond_timedwait(&sync_condition, &sync_mutex, &ts);
+		if (rv == 0)
+		{
+			WalInfo("Received connection online event signal for retrying sync notification.\n");
+		} 
+		else if (rv == ETIMEDOUT) 
+		{
+			WalPrint("SyncNotifyRetry thread: Timeout occurred.\n");
+		} 
+		else 
+		{
+			WalError("SyncNotifyRetry thread: Error occurred: %d\n", rv);	
+			continue;
+		}
+		dbCMC = getParameterValue(PARAM_CMC);
```

Comment:
We need to add a logic to avoid dbCMC get call when CMC is already 512 .
set one sync_retry_neeeded flag whenever new notification is generated and do CMC get query when this flag is enabled, if flag is false keep the thread active with 7s delay.

Reply:
Handled by using new flag g_checkSyncNotifyRetry .